lib/SPIRV/libSPIRV/SPIRVDecorate.cpp

```diff
@@ -91,6 +91,8 @@ void SPIRVDecorate::encode(spv_ostream &O) const {
     SPIRVDecorateLinkageAttr::encodeLiterals(Encoder, Literals);
   else if (Dec == DecorationMemoryINTEL)
     SPIRVDecorateMemoryINTELAttr::encodeLiterals(Encoder, Literals);
+  else if (Dec == DecorationUserSemantic)
```

Comment:
I think it is better to refactor this if-else chain into swtich...case statement